v2/gcs-to-sourcedb/src/main/java/com/google/cloud/teleport/v2/templates/processing/handler/GCSToSourceStreamingHandler.java

```diff
@@ -89,6 +112,47 @@ public static String process(ProcessingContext taskContext, SpannerDao spannerDa
     return fileProcessedStartInterval;
   }
 
+  private static void writeFilteredEventsToGcs(
+      ProcessingContext taskContext,
+      Storage storage,
+      List<TrimmedShardedDataChangeRecord> filteredEvents) {
+    Matcher matcher = Pattern.compile(GCS_INPUT_DIRECTORY_REGEX).matcher(taskContext.getGCSPath());
+
+    if (!matcher.matches()) {
+      throw new IllegalArgumentException(
+          "--gcsInputDirectoryPath is expected in a format matching "
+              + "the following regular expression: "
+              + GCS_INPUT_DIRECTORY_REGEX);
+    }
+    String bucket = matcher.group(1);
+    String path = matcher.group(2);
+    if (!path.endsWith("/")) {
+      path += "/";
+    }
+
+    String fileStartTime = taskContext.getStartTimestamp();
+    com.google.cloud.Timestamp startTs = com.google.cloud.Timestamp.parseTimestamp(fileStartTime);
+    currentIntervalStart = new org.joda.time.Instant(startTs.toSqlTimestamp());
+    org.joda.time.Instant currentIntervalEnd =
+        currentIntervalStart.plus(taskContext.getWindowDuration());
+    String gcsFileName =
+        path
+            + "filteredEvents/"
+            + taskContext.getShard().getLogicalShardId()
+            + "/"
+            + currentIntervalStart
+            + "-"
+            + currentIntervalEnd
+            + "-pane-0-last-0-of-1.txt";
```

Comment:
this text seems a little arbitrary. Is there a reason for this structure ? If yes, can you add a comment here ?

Reply:
Will add a comment. This is the same file name format as used for writing the records from spanner to GCS